src/citrine/resources/data_concepts.py

```diff
@@ -65,25 +75,25 @@ def from_dict(cls, d: dict):
         Build a data concepts object from a dictionary.
 
         This is an internal method, and should not be called directly by users.  First,
-        it removes audit_info from d, if present, and then calls the taurus object's from_dict
-        method.  Finally, it adds audit_info back into the object.
+        it removes client_specific_fields from d, if present, and then calls the taurus
+        object's from_dict method.  Finally, it adds those fields back.
 
         Parameters
         ----------
         d: dict
             A representation of the object that will be shallowly loaded into the object.
 
         """
-        audit_info = d.pop("audit_info", None)
+        popped = {k: d.pop(k, None) for k in cls.client_specific_fields}
         obj = super().from_dict(d)
-        obj.skip.add("_audit_info")
 
-        if audit_info is None:
-            obj._audit_info = None
-        elif isinstance(audit_info, dict):
-            obj._audit_info = AuditInfo.build(audit_info)
-        else:
-            raise TypeError("Audit Info must be a dictionary or an AuditInfo object")
+        for field, clazz in cls.client_specific_fields.items():
+            if popped[field] is None:
+                setattr(obj, "_{}".format(field), None)
+            elif isinstance(popped[field], dict):
+                setattr(obj, "_{}".format(field), clazz.build(popped[field]))
```

Comment:
I think we probably want to fall back on `clazz(popped[field])` when `hasattr(clazz, 'build')` is false. That would allow us to use primitive types.

Reply:
Actually there's probably a neater way than `hasattr`. Either way I think we can probably use that and ditch the is instance of `dict` check above.

Reply:
`issubclass` seems to be the thing I'm thinking of instead of `hasattr`

Reply:
Can we cross that bridge when we get there?  We'll have more information when we next decide to add a client_specific_field.

Reply:
Sure